app/assets/javascripts/discourse/components/choose-topic.js.es6

```diff
@@ -4,14 +4,41 @@ import Component from "@ember/component";
 import discourseDebounce from "discourse/lib/debounce";
 import { searchForTerm } from "discourse/lib/search";
 import { observes } from "discourse-common/utils/decorators";
+import discourseComputed from "discourse-common/utils/decorators";
 
 export default Component.extend({
   loading: null,
   noResults: null,
   topics: null,
   selectedTopicId: null,
   currentTopicId: null,
+  additionalFilters: null,
   topicTitle: null,
+  label: null,
+  loadOnInit: false,
+  topicChangedCallback: null,
+
+  init() {
+    this._super(...arguments);
+
+    this.additionalFilters = this.additionalFilters || "";
+    this.topicTitle = this.topicTitle || "";
+
+    if (this.loadOnInit && !isEmpty(this.additionalFilters)) {
+      searchForTerm(this.additionalFilters, {}).then(results => {
+        if (results && results.posts && results.posts.length > 0) {
+          this.set(
+            "topics",
+            results.posts
+              .mapBy("topic")
+              .filter(t => t.id !== this.currentTopicId)
```

Comment:
Can you access "this" here? I think the code should be similar to: https://github.com/discourse/discourse/blob/master/app/assets/javascripts/discourse/components/choose-topic.js.es6#L92